src/App.tsx

```diff
@@ -25,49 +23,67 @@ const getPolygonCovering = (
   }
   const loop = new s2.Loop(points);
   loop.normalize();
-  const s2poly = s2.Polygon.fromOrientedLoops([loop]);
-  return regionCoverer!.covering(s2poly);
+  return s2.Polygon.fromOrientedLoops([loop]);
+};
+
+const rectBuilder = (polygon: Polygon): s2.Rect => {
+  const coords = polygon.coordinates[0]; // TODO assume only the first ring
+  const latLo = Math.min(coords[0][1], coords[2][1]);
+  const lngLo = Math.min(coords[0][0], coords[2][0]);
+  const latHi = Math.max(coords[0][1], coords[2][1]);
+  const lngHi = Math.max(coords[0][0], coords[2][0]);
+
+  const DEGREE = Math.PI / 180;
+  return new s2.Rect(
+    new r1.Interval(latLo * DEGREE, latHi * DEGREE),
+    s1.Interval.fromEndpoints(lngLo * DEGREE, lngHi * DEGREE),
+  );
 };
 
-const getPolygonsCovering = (
+const getCovering = (
   regionCoverer: s2.RegionCoverer,
   polygons: Polygon[],
+  regionBuilder: { (polygon: Polygon): s2.Region },
 ): s2.CellUnion => {
   return s2.CellUnion.fromUnion(
-    ...polygons.map((p) => getPolygonCovering(regionCoverer, p)),
+    ...polygons.map((p) => regionCoverer!.covering(regionBuilder(p))),
   );
 };
 
 const getCellVisualization = (union: s2.CellUnion): FeatureCollection => {
   const degrees = s1.angle.degrees;
-  let features = union.map((cellid) => {
+  let collections = [...union].map((cellid) => {
     const cell = s2.Cell.fromCellID(cellid);
     const v0 = s2.LatLng.fromPoint(cell.vertex(0));
     const v1 = s2.LatLng.fromPoint(cell.vertex(1));
     const v2 = s2.LatLng.fromPoint(cell.vertex(2));
     const v3 = s2.LatLng.fromPoint(cell.vertex(3));
-    return {
-      type: "Feature" as "Feature",
-      geometry: {
-        type: "Polygon" as "Polygon",
-        coordinates: [
-          [
-            [degrees(v0.lng), degrees(v0.lat)],
-            [degrees(v1.lng), degrees(v1.lat)],
-            [degrees(v2.lng), degrees(v2.lat)],
-            [degrees(v3.lng), degrees(v3.lat)],
-            [degrees(v0.lng), degrees(v0.lat)],
-          ],
-        ],
-      },
-      properties: {
-        level: 0,
-      },
-    };
+
+    const p0 = point([degrees(v0.lng), degrees(v0.lat)]);
+    const p1 = point([degrees(v1.lng), degrees(v1.lat)]);
+    const p2 = point([degrees(v2.lng), degrees(v2.lat)]);
+    const p3 = point([degrees(v3.lng), degrees(v3.lat)]);
+
+    const level = s2.cellid.level(cellid);
+    const npoints = (30 - level) * 5;
+    const arc0 = greatCircle(p0, p1, { npoints });
```

Comment:
There is great-circle routine in terra-draw: 

https://github.com/JamesLMilner/terra-draw/blob/390e8051065d93de9b7effe08f96402492bca631/src/geometry/shape/great-circle-coordinates.ts

can we get away with not needing Turf? @jameslmilner